lib/async_hooks.js

```diff
@@ -476,6 +476,31 @@ function init(asyncId, type, triggerAsyncId, resource) {
   processing_hook = false;
 }
 
+// Integration with V8 Inspector AsyncTask* API
+
+const inspectorHook = createHook({
+  init(asyncId, type, triggerId, resource) {
+    // TODO(bajtos): support recurring tasks like setInterval
+    const recurring = false;
+    async_wrap.asyncTaskScheduled(type, asyncId, recurring);
+  },
+
+  before(asyncId) {
+    async_wrap.asyncTaskStarted(asyncId);
+  },
+
+  after(asyncId) {
+    async_wrap.asyncTaskFinished(asyncId);
+  },
+
+  destroy(asyncId) {
+    async_wrap.asyncTaskCanceled(asyncId);
+  },
+});
+
+// TODO(bajtos): enable the hook when Inspector is started,
+// disable it when Inspector is stopped
+inspectorHook.enable();
```

Comment:
IMO, the hook should be always enabled when debugging with no public API to control this hook. By default, the hook should be disabled. When the Inspector is started, the hook should be enabled. When the Inspector is stopped, the hook should be disabled.

The reason why this is not implemented in my patch yet, is that I haven't figured out how to make that happen.

Here is my current thinking:

 - I'll move the hook to `lib/internal/inspector-async-hook.js` and export the hook returned by `createHook`.
 - Then I need to access this exported hook (a javascript object) from `src/inspector_agent.cc` - how to do this?
 - `Agent::Start` should call `enable` on the exported hook
 - `Agent::Stop` should call `disable` on the exported hook

Thoughts?